src/wallet/wallet2.cpp

```diff
@@ -3261,8 +3262,27 @@ bool wallet2::load_unsigned_tx(const std::string &unsigned_filename, unsigned_tx
   }
   catch (...)  
   {
-    LOG_PRINT_L0("Failed to parse data from " << unsigned_filename);
-    return false;
+    LOG_PRINT_L0("Failed to parse data from " << unsigned_filename << " assuming no encryption; now trying to decrypt");
+    try
+    {
+      s = decrypt_with_view_secret_key(s);
+      try
+      {
+        std::istringstream iss(s);
+        boost::archive::portable_binary_iarchive ar(iss);
+        ar >> exported_txs;
+      }
+      catch (...)
+      {
+        LOG_PRINT_L0("Failed to parse decrypted data from " << unsigned_filename);
+        return false;
+      }
+    }
+    catch (const std::exception &e)
+    {
+      fail_msg_writer() << "Failed to decrypt " << unsigned_filename << ": " << e.what();
+      return false;
+    }
```

Comment:
This is mixing logs and message writers. I'd use logs in wallet2 since it's also used by the GUI, etc. Also this file has a version, so you don't need to try several things, you can check the version and try one based on the version (or just reject old versions for something like this, where a single person controls both ends).

Reply:
Regarding logging, I was kind of aware it didn't look OK, but I made it so because I copied some code sections from the same wallet2.cpp; i.e. `fail_msg_writer` is used in quite a few places. I guess eventually they should all be replaced by `LOG_ERROR` etc.

Regarding file versioning, I think it's important to retain the ability to read old versioned files, since some users may not be able to upgrade the wallet software installed in the computer storing the cold wallet. I've added a version checking code, but I'm not too confident if I did it right. Your comment is appreciated.